feat: add Arabic language support with RTL layout

0xpanadol opened the pull request:
- Add ar-SA locale to AvailableLocale type and i18n configuration
- Create comprehensive Arabic translations for app, public, and assets
- Implement RTL (Right-to-Left) text direction support with CSS utilities
- Add RTL detection utilities and React hooks
- Update root layout to dynamically apply dir and lang attributes
- Enhance preferences system to sync locale changes with i18n
- Add cookie-based locale persistence for translation loaders
- Include Arabic language option in preferences dropdown
- Add error handling for worker context locale switching

Closes: Arabic language support request

jurerotar commented:
I've looked over the PR and given it some thought. The RTL and localization cookie changes appear fine, since I want to support both directions and different localization, so thank you for that!

What I'm not inclined to accept are the actual localizations. The reason is that localization get updated weekly as we release new features and I have no way of keeping non-English localizations up to date. If it's okay with you, I'd love to continue with the technical changes and skip the new localizations.

I've removed the localization contribution section from the readme as well, I apologize for the oversight.

0xpanadol commented:
> I've looked over the PR and given it some thought. The RTL and localization cookie changes appear fine, since I want to support both directions and different localization, so thank you for that!
> 
> What I'm not inclined to accept are the actual localizations. The reason is that localization get updated weekly as we release new features and I have no way of keeping non-English localizations up to date. If it's okay with you, I'd love to continue with the technical changes and skip the new localizations.
> 
> I've removed the localization contribution section from the readme as well, I apologize for the oversight.

yes i agree with the part of "skip the new localizations" until the game is acutally done

jurerotar commented:
Alright, I've rebased the PR, fixed the pipeline issues and refactored it a little bit. The support for new languages works, albeit it's still scoped to a specific game world. I plan on refactoring this in the future, where this setting will not be server-based, but device based. Until then, I'll leave it as is. 

There's a couple of issues with RTL. I'm not an expert on it, so I'll ask for your opinion here.

<img width="60" height="50" alt="{78B7DFE8-946B-42DD-BFD6-11EB55F78D58}" src="https://github.com/user-attachments/assets/c5e076d5-6530-4574-abd1-6099ae0ef6b9" />

<img width="471" height="60" alt="{92D91EAD-6CFF-47F7-B992-4E88C078DC9F}" src="https://github.com/user-attachments/assets/ec092de3-a754-4257-838a-4b9b961362ff" />

If you'd like to contribute further, I welcome you to take a shot at fixing these layout issues. Since only English remains for the moment, you can enable RTL by adding `en` language to `RTL_LANGUAGE_CODES ` array.

```ts
const RTL_LANGUAGE_CODES = [
  'en', <-- Add English here
  'ar', // Arabic
  'dv', // Divehi
  'fa', // Persian (Farsi)
  'he', // Hebrew
  'ku', // Kurdish (Sorani - Arabic script)
  'ps', // Pashto
  'sd', // Sindhi
  'ug', // Uyghur
  'ur', // Urdu
  'yi', // Yiddish
];
```

Thank you for your contribution!

0xpanadol commented:
> Alright, I've rebased the PR, fixed the pipeline issues and refactored it a little bit. The support for new languages works, albeit it's still scoped to a specific game world. I plan on refactoring this in the future, where this setting will not be server-based, but device based. Until then, I'll leave it as is.
> 
> There's a couple of issues with RTL. I'm not an expert on it, so I'll ask for your opinion if these layouts look correct.
> 
> <img alt="{78B7DFE8-946B-42DD-BFD6-11EB55F78D58}" width="60" height="50" src="https://private-user-images.githubusercontent.com/28565137/475689668-c5e076d5-6530-4574-abd1-6099ae0ef6b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ1ODY3NDIsIm5iZiI6MTc1NDU4NjQ0MiwicGF0aCI6Ii8yODU2NTEzNy80NzU2ODk2NjgtYzVlMDc2ZDUtNjUzMC00NTc0LWFiZDEtNjA5OWFlMGVmNmI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA3VDE3MDcyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTllZTQ2YTg2NWVmMWE0YmNlYzhkMDE4YTQzNGE3M2ExNGMwOTkxODIxYjYwM2ZiMjA4MTNlMzA0YjZiMWVkZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WEKWnxgEbI8UJ0FV4l6xYOJuXWljVMrIvyyR7hC4TjY"> <img alt="{92D91EAD-6CFF-47F7-B992-4E88C078DC9F}" width="471" height="60" src="https://private-user-images.githubusercontent.com/28565137/475689844-ec092de3-a754-4257-838a-4b9b961362ff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ1ODY3NDIsIm5iZiI6MTc1NDU4NjQ0MiwicGF0aCI6Ii8yODU2NTEzNy80NzU2ODk4NDQtZWMwOTJkZTMtYTc1NC00MjU3LTgzOGEtNGI5Yjk2MTM2MmZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA3VDE3MDcyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM4NDQ4Y2JiMmM1MDg5YTBmM2E1OTEwYTBhZjlhYmI4NDQ5NGU1MjBlNTU3ZWQwNDZhZjQ2NzY1M2JkYTMzMWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hyghWPYXtPp8arHMXZwT_-4eWzDIOfTDukfAcNvq7S8"> <img alt="{2CC3A38E-6E75-418F-95CA-0D32B9010DAB}" width="1275" height="237" src="https://private-user-images.githubusercontent.com/28565137/475691985-1a1eb40c-9f37-4a6f-b78b-68ff938ba3f9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ1ODY3NDIsIm5iZiI6MTc1NDU4NjQ0MiwicGF0aCI6Ii8yODU2NTEzNy80NzU2OTE5ODUtMWExZWI0MGMtOWYzNy00YTZmLWI3OGItNjhmZjkzOGJhM2Y5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA3VDE3MDcyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1ZDYzMWY1M2JlNTY5MTMyNDc3OTJlZWU2OWRiOWFjNjY3YWJmNjdlZTRjODBmMDQ1YjZjOTVlZmZjOWM0ZTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.saQ-yO-9Y0ai4Nhwhfu-rec6vM2PCTBZl1BGvNkAP9A">
> If you'd like to contribute further, I welcome you to take a shot at fixing these layout issues. Since only English remains for the moment, you can enable RTL by adding `en` language to `RTL_LANGUAGE_CODES ` array.
> 
> ```ts
> const RTL_LANGUAGE_CODES = [
>   'en', <-- Add English here
>   'ar', // Arabic
>   'dv', // Divehi
>   'fa', // Persian (Farsi)
>   'he', // Hebrew
>   'ku', // Kurdish (Sorani - Arabic script)
>   'ps', // Pashto
>   'sd', // Sindhi
>   'ug', // Uyghur
>   'ur', // Urdu
>   'yi', // Yiddish
> ];
> ```
> 
> Thank you for your contribution!

Thanks for your work.
